TransferMethodRepository/Sources/TransferMethodRepository.swift

```diff
@@ -0,0 +1,139 @@
+}
+
+final class RemoteTransferMethodRepository: TransferMethodRepository {
+    private static let deactivateNote = "Deactivating Account"
+
+    func create(_ transferMethod: HyperwalletTransferMethod,
+                _ completion: @escaping (Result<HyperwalletTransferMethod?, HyperwalletErrorType>) -> Void) {
+        if let bankAccount = transferMethod as? HyperwalletBankAccount {
+            Hyperwallet.shared.createBankAccount(account: bankAccount, completion: createHandler(completion))
+        } else if let bankCard = transferMethod as? HyperwalletBankCard {
+            Hyperwallet.shared.createBankCard(account: bankCard, completion: createHandler(completion))
+        } else if let payPalAccount = transferMethod as? HyperwalletPayPalAccount {
+            Hyperwallet.shared.createPayPalAccount(account: payPalAccount, completion: createHandler(completion))
+        } else {
+            logTransferMethodTypeNotSupported(transferMethod.type ?? "")
+        }
+    }
+
+    func deactivate(_ transferMethod: HyperwalletTransferMethod,
+                    _ completion: @escaping (Result<HyperwalletStatusTransition?, HyperwalletErrorType>) -> Void) {
+        guard let transferMethodType = transferMethod.type,
+            let token = transferMethod.token else {
+            return
+        }
+
+        switch transferMethodType {
+        case "BANK_ACCOUNT", "WIRE_ACCOUNT":
+            Hyperwallet.shared.deactivateBankAccount(transferMethodToken: token,
+                                                     notes: RemoteTransferMethodRepository.deactivateNote,
+                                                     completion: deactivateHandler(completion))
+        case "BANK_CARD":
+            Hyperwallet.shared.deactivateBankCard(transferMethodToken: token,
+                                                  notes: RemoteTransferMethodRepository.deactivateNote,
+                                                  completion: deactivateHandler(completion))
+        case "PAYPAL_ACCOUNT":
+            Hyperwallet.shared.deactivatePayPalAccount(transferMethodToken: token,
+                                                       notes: RemoteTransferMethodRepository.deactivateNote,
+                                                       completion: deactivateHandler(completion))
+
+        default:
+            logTransferMethodTypeNotSupported(transferMethodType)
+        }
+    }
+
+    func list(
+        _ queryParam: HyperwalletTransferMethodQueryParam?,
+        _ completion: @escaping (Result<HyperwalletPageList<HyperwalletTransferMethod>?,
+                                        HyperwalletErrorType>) -> Void) {
+        Hyperwallet.shared.listTransferMethods(queryParam: queryParam,
+                                               completion: listTransferMethodHandler(completion))
+    }
+
+    private func createHandler(
```

Comment:
I think you can get rid of all the specific handlers, and instead you can use a generic handler. Because the handlers logic are exactly the same.  Something like:
```

    private func completionHandler<T>(
        _ completion: @escaping (Result<T?, HyperwalletErrorType>) -> Void)
        -> (T?, HyperwalletErrorType?) -> Void {
            return {(result, error) in
                CompletionHelper.performCompletion(error, result, completion)
            }
    }
```

Reply:
Good idea.